### PR TITLE
feature/BNGP-4369-ITHC-csp-fix

### PR DESCRIPTION
**Pull request by soorajvdefra**

### CODE REVIEW

**Review by pwadmore-ea — CHANGES_REQUESTED** [review]
In addition to my comments on individual files I recommend adding some unit tests to check the content security policy header is being set correctly. For example, https://github.com/DEFRA/wildlife-licencing/blob/29a748ef8d15c1fe7981ea8da3ad570c1987a1d0/packages/web-service/src/__tests__/additional-page-data.spec.js

File: packages/webapp/src/server.js [hunk 1]
```diff
@@ -46,6 +47,35 @@ const init = async server => {
...
+      }
+
+      // Set the nonce in the context
+      response.source.context.nonce = nonce
+
+      // Set CSP header if response allows setting headers
+      if (typeof response.header === 'function') {
+        const scriptHash = '+6WnXIl4mbFTCARd8N3COQmT3bJJmo32N8q8ZSQAIcU='
+        const csp = 'default-src \'self\'; ' +
+                        `script-src 'self' 'nonce-${nonce}' 'sha256-${scriptHash}'; ` +
```
[the remark this answers is not in the document]
Comment (soorajvdefra) [reply packages/webapp/src/server.js:69]: hi @pwadmore-ea i have used nonce only 'sha256 also required to fix the internal inline script of govuk/template.njk
reference 
https://github.com/alphagov/govuk-frontend/issues/1657
https://technology.blog.gov.uk/2015/02/12/experimenting-with-content-security-policy-on-gov-uk/

[the remark this answers is not in the document]
Comment (soorajvdefra) [reply packages/webapp/src/server.js]: comment and gtm  added

File: packages/webapp/src/server.js [hunk 2]
```diff
@@ -46,6 +47,35 @@ const init = async server => {
...
+      // Ensure the context object exists for the view
+      if (!response.source.context) {
+        response.source.context = {}
+      }
+
+      // Set the nonce in the context
+      response.source.context.nonce = nonce
+
+      // Set CSP header if response allows setting headers
+      if (typeof response.header === 'function') {
```
[the remark this answers is not in the document]
Comment (soorajvdefra) [reply packages/webapp/src/server.js]: this code removed as its not required

**Review by pwadmore-ea — APPROVED** [review]

**Comment by shaundefra** [comment]
tested locally

**Pull request merged**